ConcernsCaseWork/ConcernsCaseWork/Pages/Case/Management/Action/SRMA/Edit/DateOfVisit.cshtml.cs

```diff
@@ -0,0 +1,144 @@
+﻿using ConcernsCaseWork.Helpers;
+using ConcernsCaseWork.Models.CaseActions;
+using ConcernsCaseWork.Pages.Base;
+using ConcernsCaseWork.Services.Cases;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Threading.Tasks;
+
+namespace ConcernsCaseWork.Pages.Case.Management.Action.SRMA
+{
+	[Authorize]
+	[ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+	public class EditDateOfVisitPageModel : AbstractPageModel
+	{
+		private readonly ISRMAService _srmaModelService;
+		private readonly ILogger<EditDateOfVisitPageModel> _logger;
+
+		public SRMAModel SRMAModel { get; set; }
+
+		public EditDateOfVisitPageModel(ISRMAService srmaModelService, ILogger<EditDateOfVisitPageModel> logger)
+		{
+			_srmaModelService = srmaModelService;
+			_logger = logger;
+		}
+
+		public async Task<ActionResult> OnGetAsync()
+		{
+			long caseUrn = 0;
+			long srmaId = 0;
+
+			try
+			{
+				_logger.LogInformation("Case::Action::SRMA::EditDateOfVisitPageModel::OnGetAsync");
+
+				(caseUrn, srmaId) = GetRouteData();
+
+				SRMAModel = await _srmaModelService.GetSRMAById(srmaId);
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError("Case::Action::SRMA::EditDateOfVisitPageModel::OnGetAsync::Exception - {Message}", ex.Message);
+				TempData["Error.Message"] = ErrorOnGetPage;
+			}
+
+			return Page();
+		}
+
+		public async Task<ActionResult> OnPostAsync()
+		{
+			long caseUrn = 0;
+			long srmaId = 0;
+
+			try
+			{
+				_logger.LogInformation("Case::Action::SRMA::EditDateOfVisitPageModel::OnPostAsync");
+
+				(caseUrn, srmaId) = GetRouteData();
+
+				SRMAModel = await _srmaModelService.GetSRMAById(srmaId);
+
+				var start_dtr_day = Request.Form["start-dtr-day"].ToString();
+				var start_dtr_month = Request.Form["start-dtr-month"].ToString();
+				var start_dtr_year = Request.Form["start-dtr-year"].ToString();
+
+				var end_dtr_day = Request.Form["end-dtr-day"].ToString();
+				var end_dtr_month = Request.Form["end-dtr-month"].ToString();
+				var end_dtr_year = Request.Form["end-dtr-year"].ToString();
+
+				var start_dtString = $"{start_dtr_day}-{start_dtr_month}-{start_dtr_year}";
+				var end_dtString = string.IsNullOrEmpty(end_dtr_day) && string.IsNullOrEmpty(end_dtr_month) && string.IsNullOrEmpty(end_dtr_year) ? 
+					String.Empty : $"{end_dtr_day}-{end_dtr_month}-{end_dtr_year}";
+
+
+				ValidateDatesOfVisit(start_dtString, end_dtString);
+				var visitDates = ParseVisitDates(start_dtString, end_dtString);
+
+				await _srmaModelService.SetVisitDates(srmaId, visitDates.startDate, visitDates.endDate);
+				return Redirect($"/case/{caseUrn}/management/action/srma/{srmaId}");
+			}
+			catch (ApplicationException ex)
```

Comment:
It's not recommended to throw or derive from ApplicationException. can you please use a different exception (create our own, use invalidoprationex, etc) or we can use a more straightforward approach to deliver feedback to user.